ols/app/main.py

```diff
@@ -53,4 +54,60 @@ async def rest_api_counter(
     return response
 
 
+@app.middleware("")
+async def log_requests_responses(
+    request: Request, call_next: Callable[[Request], Awaitable[Response]]
+) -> Response:
+    """Middleware for logging of HTTP requests and responses, at debug level."""
+    if logger.isEnabledFor(logging.DEBUG):
+        # Log request headers
+        request_log_message = (
+            f"Request {request.client.host}:{request.client.port}"
+            f" {request.headers}, Body: "
+        )
+
+        # Log request body
+        request_body = await request.body()
+        if request_body:
+            request_log_message += f"{request_body.decode('utf-8')}"
+        else:
+            request_log_message += "None"
+
+        logger.debug(request_log_message)
+
+    # Process request
+    response = await call_next(request)
+
+    if logger.isEnabledFor(logging.DEBUG):
```

Comment:
we can return response here to use only one checking on logger.isEnabledFor(logging.DEBUG)
```
if not logger.isEnabledFor(logging.DEBUG): 
  return response
```

Reply:
I'm not sure what you mean. I don't think we can get away with just a single check, we need to produce and return a response even if we're not logging
```
if logger.isEnabledFor(logging.DEBUG):
    # log request
response = await call_next(request)
if logger.isEnabledFor(logging.DEBUG):
    # log response
return response
```

Reply:
the current version is good. I mean that it seems more clear to me if the code is arranged like:
```python
if not logger.isEnabledFor(logging.DEBUG): 
    return call_next(request) 
# log request 
response = await call_next(request) 
# log response 
return response
```
I like the less indented layout and put the most probable code path at the beginning.

Reply:
Got it, thanks!